### PR TITLE
Fix for passed axes handling in plot_posterior

### LINKED ISSUE #1827: plot_posterior assumes iterable of axes for ax argument
When passing custom axes to `plot_posterior`, the function assumes the `ax` argument is an iterable, which ought not to be assumed:

```
fig, axes = plt.subplots(figsize=(12,8))
plot_posterior(trace, ax=axes)
```

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-14-435af7bcc2a7> in <module>()
      1 fig, axes = plt.subplots(figsize=(12,8))
----> 2 plot_posterior(trace, ax=axes)

/Users/fonnescj/Repos/pymc3/pymc3/plots.py in plot_posterior(trace, varnames, transform, figsize, alpha_level, round_to, point_estimate, rope, ref_val, kde_plot, plot_transformed, ax, **kwargs)
    780             ax, fig = create_axes_grid(figsize, trace_dict)
    781 
--> 782         for a, v in zip(ax, trace_dict):
    783             tr_values = transform(trace_dict[v])
    784             plot_posterior_op(tr_values, ax=a)

TypeError: zip argument #1 must support iteration
```

However, even putting the axes in a list results in an error:

```
fig, axes = plt.subplots(figsize=(12,8))
plot_posterior(trace, ax=[axes])
```

Looks like it uses `fig` which does not get created when axes are passed:

```
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-15-43749b0abf36> in <module>()
      1 fig, axes = plt.subplots(figsize=(12,8))
----> 2 plot_posterior(trace, ax=[axes])

/Users/fonnescj/Repos/pymc3/pymc3/plots.py in plot_posterior(trace, varnames, transform, figsize, alpha_level, round_to, point_estimate, rope, ref_val, kde_plot, plot_transformed, ax, **kwargs)
    785             a.set_title(v)
    786 
--> 787         fig.tight_layout()
    788     return ax
    789 

UnboundLocalError: local variable 'fig' referenced before assignment

```

### PR DESCRIPTION
**Pull request by fonnesbeck**
Closes #1827

### CODE REVIEW

File: pymc3/plots.py [hunk 1]
```diff
@@ -779,12 +779,12 @@ def get_trace_dict(tr, varnames):
         if ax is None:
             ax, fig = create_axes_grid(figsize, trace_dict)
 
-        for a, v in zip(ax, trace_dict):
+        for a, v in zip(np.atleast_1d(ax), trace_dict):
```
Comment (twiecki) [inline pymc3/plots.py:782]: I think ax.flatten() is commonly used. Or maybe that doesn't work here?

Comment (fonnesbeck) [reply pymc3/plots.py:782]: I'm not trying to flatten, I'm trying to ensure that a single axis passed is iterable.

**Comment by fonnesbeck** [comment]
Anyone have issues with this fix?

**Comment by twiecki** [comment]
LGTM.

**Pull request merged**